WooCommerce/Classes/Authentication/Navigation Exceptions/NoWooErrorViewModel.swift

```diff
@@ -0,0 +1,157 @@
+    }
+
+    // MARK: - Data and configuration
+    let image: UIImage = .noStoreImage
+
+    var text: NSAttributedString {
+        let font: UIFont = .body
+        let boldFont: UIFont = font.bold
+
+        let boldSiteAddress = NSAttributedString(string: siteURL.trimHTTPScheme(),
+                                                           attributes: [.font: boldFont])
+        let message = NSMutableAttributedString(string: Localization.errorMessage)
+
+        message.replaceFirstOccurrence(of: "%@", with: boldSiteAddress)
+
+        return message
+    }
+
+    var isAuxiliaryButtonHidden: Bool { !showsConnectedStores }
+
+    let auxiliaryButtonTitle = Localization.seeConnectedStores
+
+    let primaryButtonTitle = Localization.primaryButtonTitle
+
+    var isPrimaryButtonHidden: Bool { !showsInstallButton }
+
+    let secondaryButtonTitle = Localization.secondaryButtonTitle
+
+    // MARK: - Actions
+    func didTapPrimaryButton(in viewController: UIViewController?) {
+        analytics.track(.loginWooCommerceSetupButtonTapped)
+        guard let viewController = viewController else {
+            return
+        }
+        let viewModel = WooSetupWebViewModel(siteURL: siteURL, onCompletion: {
+            viewController.navigationController?.popViewController(animated: true)
+            handleSetupCompletion(in: viewController)
+        })
+        let setupViewController = PluginSetupWebViewController(viewModel: viewModel)
+        viewController.navigationController?.show(setupViewController, sender: nil)
+    }
+
+    func didTapSecondaryButton(in viewController: UIViewController?) {
+        // Log out and pop
+        stores.deauthenticate()
+        viewController?.navigationController?.popToRootViewController(animated: true)
+    }
+
+    func didTapAuxiliaryButton(in viewController: UIViewController?) {
+        guard let navigationController = viewController?.navigationController else {
+            return
+        }
+
+        let storePicker = StorePickerViewController()
+        storePicker.configuration = .listStores
+
+        navigationController.pushViewController(storePicker, animated: true)
+    }
+
+    func viewDidLoad() {
+        analytics.track(.loginWooCommerceErrorShown)
+    }
+}
+
+// MARK: - Private helpers
+private extension NoWooErrorViewModel {
+    func handleSetupCompletion(in viewController: UIViewController, retryCount: Int = 0) {
+        showInProgressView(in: viewController)
+
+        ServiceLocator.stores.synchronizeEntities {
+            viewController.navigationController?.dismiss(animated: true)
+
+            let matcher = ULAccountMatcher()
+            matcher.refreshStoredSites()
+            guard let site = matcher.matchedSite(originalURL: siteURL),
+                  site.isWooCommerceActive else {
+                if retryCount < 1 {
+                    return handleSetupCompletion(in: viewController, retryCount: retryCount + 1)
```

Comment:
I didn't test this scenario. But will we end up presenting `InProgressViewController` twice, leaving the navigation stack in an undesired state? (i.e. Popping will not produce the expected result) 🤔

Reply:
This can be tested by quickly turning internet off on the first request, then turning it back on before tapping retry 😅 The loading screen is dismissed at the start of the completion block, so we won't present it twice.

Reply:
Oh, I totally missed it. 🤦 

Also, kindly consider adding a comment to explain the dismissal.

Reply:
Done in 2e5d625eb57ea7a2b1b9c3b4a77db549551ab26b 👍